---
title: "Network analysis of the Lymphome proteomics data set"
author: "Julien Chiquet"
date: "February 2019"
output:
  pdf_document:
    toc: true
    number_section: true
  html_document:
    toc: true
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE, warnings = FALSE, fig.align="center")
```

# Requirements {-}

```{r requirement, message = FALSE}
library(tidyverse)
library(corrplot)
library(GGally)
library(igraph)
library(ggfortify)
library(blockmodels)
library(QUIC)
library(stabs)
library(simone)
library(RColorBrewer)
pal <- brewer.pal(10, "Set3") # a fancy palette for graph
```

# Basic proteomics data analysis

We first load both proteomics and covariates

```{r data loading}
proteomics <- readRDS("proteins.rds")
covariates <- readRDS("covariates.rds")
```

Let us have a quick look on both data frames:

## Covariates

First a quick look,

```{r see covariates}
covariates %>% head() %>% knitr::kable()
```

Only women are subject to this disease, so let us remove men:

```{r remove men}
proteomics <- proteomics[covariates$sex == "F", ]
covariates <- covariates %>% filter(sex == "F") %>% dplyr::select(-sex)
```

Then do a pairs plot:

```{r covariates pairs}
GGally::ggpairs(covariates, columns = 2:4, aes(colour = disease))
```

## Proteomics

Again first a quick look:

```{r see proteomics}
proteomics %>% head() %>% knitr::kable()
```


Simple PCA does not seem to show a strong structuration of the data due to the disease status:

```{r see PCA proteomics}
pca <- prcomp(proteomics, scale. = TRUE)
for_plot <- as_tibble(proteomics) %>% 
  add_column(disease = covariates$disease,
             type    = covariates$type)
autoplot(pca, loadings = TRUE, loadings.label = TRUE,
         data = for_plot, colour = 'disease')
# autoplot(pca, loadings = TRUE, loadings.label = TRUE,
#          data = for_plot, colour = 'type')
```

Still, there is some structure between the proteines:

```{r corrplot}
corrplot(cor(proteomics), order = "hclust")
```

# Tutorial: proteomics network reconstruction on pooled data

## Neighborhood selection with *simone*

We first retrieve the whole path of inferred networks with *simone*.

```{r pooled network, results = FALSE}
pooled_path <- simone(proteomics)
```

And extract one of them:

```{r singe network}
myNetwork <- getNetwork(pooled_path)
plot(myNetwork)
```

There seems to be two clusters: let us manually adjus a stochastic block model on this network

```{r adjust SBM pooled}
subNetwork <- myNetwork$A
to_remove <- rowSums(subNetwork) == 0
subNetwork <- subNetwork[!to_remove, !to_remove]
mySBM_collection <- 
  BM_bernoulli(
    "SBM_sym", 
    subNetwork, 
    verbosity = 0
  )
mySBM_collection$estimate()
```

The ICL criterion choses 3 clusters. 

```{r extract clusters single prot}
clusters <- apply(mySBM_collection$memberships[[2]]$Z, 1, which.max)
```

The following piece of code returns a fancy version of this graph thanks to the *igraph* package ready for plotting: 

```{r fancy plot igraph}
fancy_igraph <- function(graph, clusters, factor = 5) {
  V(graph)$class <- clusters
  V(graph)$size <- degree(graph) * factor
  V(graph)$frame.color <- "white"
  V(graph)$color <- pal[V(graph)$class]
  E(graph)$arrow.mode <- 0
  graph
}
```

Now, let us represent the graph from the adjacency matrix and a weighted version with partial correlation values :

```{r plot SBM igraph}
g1 <- graph_from_adjacency_matrix(subNetwork, mode = "undirected", diag = FALSE)

partialCor <- -myNetwork$Theta
partialCor <- partialCor[!to_remove, !to_remove]

g2 <- graph_from_adjacency_matrix(partialCor, weighted = TRUE, mode = "undirected", diag = FALSE)

par(mfrow = c(1,2))
plot(fancy_igraph(g1, clusters), main = "Adjacency Network")
plot(fancy_igraph(g2, clusters), edge.width = 20*E(g2)$weight, main = "Partial Correlations")
```

## Graphical-Lasso with Stability Selection

The graphical-Lasso is efficiently implemented by the QUIC algorithm. An alternative to BIC to choose the most stables edges in the network is the stability selection approach. It subsamples the original data set, run the algorithm on each subsample and estimate the probability of selection of each edge along the path.


```{r stabsel + graphical lasso, cache = TRUE, warning=FALSE, message = FALSE}
Glasso_stability <- stabsel(x = proteomics, fitfun = "quic.graphical_model", cutoff = 0.985, PFER = 5)
```

```{r stabsel plot}
plot(Glasso_stability, type = "maxsel", labels = 1:50, main = "selection") 
```

```{r extract Glasso net}
g_stabs <- 
  graph_from_edgelist(
    do.call(rbind, strsplit(names(Glasso_stability$selected), " : ")),
    directed = FALSE
  )
```

Now try to find some structure thanks to blockmodels:
```{r SBM Glasso}
mySBM_collection <- 
  BM_bernoulli(
    "SBM_sym", 
    as_adjacency_matrix(g_stabs, sparse = FALSE),
    verbosity = 0
  )
mySBM_collection$estimate()
clusters <- apply(mySBM_collection$memberships[[2]]$Z, 1, which.max)
```

And plot it:

```{r plot Glasso stabs}
plot(fancy_igraph(g_stabs, clusters))
```

What is the intersection with plot inferred with neighborhood selection and *simone*?

```{r intersection}
consensus_graph <- igraph::intersection(g_stabs, g2)
mySBM_collection <- 
  BM_bernoulli(
    "SBM_sym", 
    as_adjacency_matrix(consensus_graph, sparse = FALSE),
    verbosity = 0, plotting = ""
  )
mySBM_collection$estimate()
clusters <- apply(mySBM_collection$memberships[[2]]$Z, 1, which.max)
plot(
  delete.vertices(
    fancy_igraph(consensus_graph, clusters),
    degree(consensus_graph) == 0
    )
  )
```


# Practical: network inference accounting for patient status

## Consensus and differential network from independent network inference 

  - Infer one network per patient status, with a method of your choice (e.g. *stabsel* with 'quic' fit function, or neighborhood selection with *huge* and STaRS for model selection)
  - Compare these two networks according to
    - some descriptive statistics learnt during today's course
    - the clusterings obtained with *blockmodels*
  - Build a  "differential" and a "consensus" network from the two network associated with each status
  
## Robustness of the networks

Let us try a "home-made" stability selection for evaluating the robustness of the differential or consensus network: For each patient status, 
  - create $K$ subsamples with, says 90% of the original data
  - repeat the analysis described in 2.1
  - evaluate the stability of the differential and consensus networks

## Consensus network with joint inference

  - Use the *simone* multitask framework to infer the networks jointly
  - Try to find some edges specific to each patient status
  - Compare your results with the first approach (that consider network inference independently)
  - If you have time (and multiple cores on your computer...), evaluate the robustness of the consensus network with resampling
  
## Optional 

### Weighted penalties

You may put some prior in the network inference model by weighting the penalty associated to each edge:

To this end, build a matrix of weights which depends on the connection probabilities inferred by the SBM ($\varpropto (1 - \hat{\pi}_{ij})$) to refine the inferred networks (parameter $rho$ in function 'quic.graphical_model'). Connection probabilities can be extracted from a collection of *blockmodels* objects via `SBM_collection$model_parameters[[1]]$pi`.

### Multiattribute network from proteomics and transcriptomics

You may find a consensus network by coupling transcriptomics and proteomics data: first determine the transcripts associated with the set of 20 proteins. Then use the package found here:

```{r multivarNetwork install, eval = FALSE}
devtools::install_github("jchiquet/multivarNetwork")
```
